cmd/goal/asset.go

```diff
@@ -754,42 +763,66 @@ var infoAssetCmd = &cobra.Command{
 		accountList := makeAccountsList(dataDir)
 		creator := accountList.getAddressByName(assetCreator)
 
+		// Helper methods for dereferencing optional asset fields.
+		derefString := func(s *string) string {
+			if s == nil {
+				return ""
+			}
+			return *s
+		}
+		derefBool := func(b *bool) bool {
+			if b == nil {
+				return false
+			}
+			return *b
+		}
+
 		lookupAssetID(cmd, creator, client)
 
-		params, err := client.AssetInformation(assetID)
+		asset, err := client.AssetInformationV2(assetID)
 		if err != nil {
 			reportErrorf(errorRequestFail, err)
 		}
 
 		reserveEmpty := false
-		if params.ReserveAddr == "" {
+		if derefString(asset.Params.Reserve) == "" {
 			reserveEmpty = true
-			params.ReserveAddr = params.Creator
+			asset.Params.Reserve = &asset.Params.Creator
 		}
 
-		reserve, err := client.AccountInformation(params.ReserveAddr)
+		reserve, err := client.AccountInformationV2(*asset.Params.Reserve, true)
```

Comment:
I think `client.AccountAssetInformation(*asset.Params.Reserve, assetID)` could be used instead of `client.AccountInformationV2`

Reply:
Followed up offline: It looks like we'd want to change the behavior so that `goal asset info` fails if the reserve doesn’t hold the asset - this will likely involve changing a test (`test/scripts/e2e_subs/asset-misc.sh`) to address this in a follow-up PR.